cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetPanel/index.tsx

```diff
@@ -0,0 +1,100 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Button from '@material-ui/core/Button';
+import withStyles, { StyleRules } from '@material-ui/core/styles/withStyles';
+import { useWidgetState } from 'components/PluginJSONCreator/Create';
+import WidgetActionButtons from 'components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetActionButtons';
+import WidgetAttributesCollection from 'components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection';
+import WidgetInfoInput from 'components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetInfoInput';
+import * as React from 'react';
+
+const styles = (): StyleRules => {
+  return {
+    eachWidget: {
+      display: 'grid',
+      gridTemplateColumns: '5fr 1fr',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+    },
+  };
+};
+
+export const WidgetPanelView = ({
+  classes,
+  widgetID,
+  widgetIndex,
+  openWidgetIndex,
+  setOpenWidgetIndex,
+  addWidgetToGroup,
+  deleteWidgetFromGroup,
+  onWidgetNameChange,
+  onWidgetLabelChange,
+  onWidgetTypeChange,
+  onWidgetCategoryChange,
+}) => {
+  function openWidgetAttributes(index: number) {
+    return () => {
+      setOpenWidgetIndex(index);
+    };
+  }
+
+  function closeWidgetAttributes() {
+    setOpenWidgetIndex(null);
+  }
+
+  const { widgetInfo, widgetToAttributes } = useWidgetState();
+
+  return React.useMemo(
+    () => (
+      <div className={classes.eachWidget}>
+        <WidgetInfoInput
+          widgetID={widgetID}
+          onWidgetNameChange={onWidgetNameChange}
+          onWidgetLabelChange={onWidgetLabelChange}
+          onWidgetTypeChange={onWidgetTypeChange}
+          onWidgetCategoryChange={onWidgetCategoryChange}
+        />
+        <WidgetActionButtons
+          onAddWidgetToGroup={addWidgetToGroup}
+          onDeleteWidgetFromGroup={deleteWidgetFromGroup}
+        />
+        <WidgetAttributesCollection
+          widgetID={widgetID}
+          widgetAttributesOpen={openWidgetIndex === widgetIndex}
+          closeWidgetAttributes={closeWidgetAttributes}
+          onWidgetNameChange={onWidgetNameChange}
+          onWidgetLabelChange={onWidgetLabelChange}
+          onWidgetTypeChange={onWidgetTypeChange}
+          onWidgetCategoryChange={onWidgetCategoryChange}
+        />
+        <Button
+          variant="contained"
+          color="primary"
+          component="span"
+          onClick={openWidgetAttributes(widgetIndex)}
+        >
+          Attributes
+        </Button>
+        hmm {JSON.stringify(widgetInfo.get(widgetID))}
```

Comment:
delete the `hmm`?

Reply:
Oh haha, forgot to delete something I wrote while debugging

Reply:
Resolved